manifests/server/instance/config.pp

```diff
@@ -219,9 +219,11 @@
     }
   }
 
-  postgresql::server::config_entry { "data_directory_for_instance_${name}":
-    key   => 'data_directory',
-    value => $datadir,
+  unless $facts['service_provider'] == 'systemd' and $facts['os']['family'] in ['RedHat', 'Gentoo'] {
+    postgresql::server::config_entry { "data_directory_for_instance_${name}":
```

Comment:
For upgrades, should we ensure it's absent on systemd + RH/Gentoo?

Reply:
Probably yes?